client/cmd/dexc-desktop/app_darwin.go

```diff
@@ -0,0 +1,584 @@
+	addMethodToDelegate("applicationWillFinishLaunching:", func(_ objc.Object) {
+		// Create the main menu bar.
+		menuBar := cocoa.NSMenu_New()
+		app.SetMainMenu(menuBar)
+		appMenus := createMainMenuItems()
+		for _, menu := range appMenus {
+			// Create a menu item for the menuBar and set the menu as the
+			// submenu. See:
+			// https://developer.apple.com/documentation/appkit/nsmenuitem/1514845-submenu?language=objc
+			mainBarItem := cocoa.NSMenuItem_New()
+			mainBarItem.SetTitle(menu.Title())
+			mainBarItem.SetSubmenu(menu)
+			menuBar.AddItem(mainBarItem)
+
+			if menu.Title() == "Window" {
+				// Set NSApp's WindowsMenu to the Window menu. This will allow
+				// windows to be grouped together in the dock icon and in the
+				// Window menu. Also, MacOS will automatically add other default
+				// Window menu items. See:
+				// https://developer.apple.com/documentation/appkit/nsapplication/1428547-windowsmenu?language=objc
+				app.Set("WindowsMenu:", menu)
+			}
+		}
+
+		// Create the status bar menu. We want users to notice dexc desktop is
+		// still running (even with the dot below the dock icon).
+		obj := cocoa.NSStatusBar_System().StatusItemWithLength(cocoa.NSVariableStatusItemLength)
+		obj.Retain()
+		obj.Button().SetImage(cocoa.NSImage_InitWithData(core.NSData_WithBytes(SymbolBWIcon, uint64(len(SymbolBWIcon)))))
+		obj.Button().Image().SetSize(core.Size(18, 18))
+		obj.Button().SetToolTip("Self-custodial multi-wallet")
+
+		runningItem := cocoa.NSMenuItem_New()
+		runningItem.SetTitle("Dex Client is running")
+		runningItem.SetEnabled(false)
+
+		itemQuit := cocoa.NSMenuItem_Init("Quit", objc.Sel("terminate:"), "q")
+		itemQuit.SetToolTip("Quit DEX client")
+
+		menu := cocoa.NSMenu_New()
+		menu.AddItem(runningItem)
+		menu.AddItem(itemQuit)
+		obj.SetMenu(menu)
+
+		// Hide the application until it is ready to be shown when we receive
+		// the "NSApplicationDidFinishLaunchingNotification" below. This also
+		// allows us to ensure the menu bar is redrawn.
+		app.TryToPerform_with_(objc.Sel("hide:"), nil)
+	})
+
+	// MacOS will always send a notification named
+	// "NSApplicationDidFinishLaunchingNotification" when an application has
+	// finished launching. See:
+	// https://developer.apple.com/documentation/appkit/nsapplicationdidfinishlaunchingnotification?language=objc
+	addMethodToDelegate("applicationDidFinishLaunching:", func(_, notification objc.Object) {
+		// Unhide the app on the main thread after it has finished launching we
+		// need to give this priority before creating the window to ensure the
+		// window is immediately visible when it's created. This also has the
+		// side effect of redrawing the menu bar which will be unresponsive
+		// until it is redrawn.
+		core.Dispatch(func() {
+			app.TryToPerform_with_(objc.Sel("unhide:"), nil)
+		})
+
+		createNewWebView()
+	})
+
+	app.SetActivationPolicy(cocoa.NSApplicationActivationPolicyRegular)
+	app.ActivateIgnoringOtherApps(false)
```

Comment:
The false value means we don't want another `dexc-desktop` app process to start if one is already running. If set to true, macOS will `ignore` other `dexc-desktop` process and attempt to start anyways. This is only true when `dexc-desktop` is started using the icon. The other process file is to prevent starting `dexc-desktop` from the terminal (which will fail anyways) when `dexc-desktop` is already running.